File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-6.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        charset = set()
        maxlength = 0
        left = 0

        for right in range(len(s)):
            while s[right] in charset:
                charset.remove(s[left])
                left += 1
            charset.add(s[right])
            
            maxlength = max(maxlength, right - left + 1)
        
        return maxlength
        

            

        

File: Data Structures & Algorithms/max-water-container/submission-6.java
class Solution {
    public int maxArea(int[] heights) {

      

        int left = 0;
        int right = heights.length - 1;
        int water = 0;


        while(left < right){

           int maxArea = Math.min(heights[left], heights[right]) * (right - left);

            water = Math.max(maxArea, water);

            if(heights[left] < heights[right]){
                left++;
            }
            else{
                right--;
            }

        }
        
        return water;
    }
}
